pkg/framework/aries/framework.go

```diff
@@ -373,6 +382,19 @@ func createOutboundDispatcher(frameworkOpts *Aries) error {
 	return nil
 }
 
+func createMessenger(frameworkOpts *Aries) error {
+	ctx, err := context.New(
+		context.WithOutboundDispatcher(frameworkOpts.outboundDispatcher),
+	)
+	if err != nil {
+		return fmt.Errorf("context creation failed: %w", err)
+	}
+
+	frameworkOpts.messenger = messenger.NewMessenger(ctx)
```

Comment:
Shouldn't Messenger be configurable ? ie, Pass it as an option while creating the framework and framework provides just a default option.

Reply:
agree - we should always have configurable components in framework.

note: the whole inbound function that is currently in context.go should be configurable...

Reply:
@troyronda @rolsonquadras  I have added a possibility to pass it as an option. But I can not remove it. We are using OutboundDispatcher for the messenger. Because of that our BBD tests are failing!

Reply:
@soluchok In the new code, the default messenger overrides the passed in option all the time. We can provide the default implementation in https://github.com/hyperledger/aries-framework-go/blob/master/pkg/framework/aries/default.go#L74.

Reply:
It is good to know, but it does not help. I have added an additional check for that. And the unit test as well. Now you can pass it as an option. https://github.com/hyperledger/aries-framework-go/pull/1098/files#diff-7b2448fdc2181299533bc944239b9744R379-R382

Reply:
Its created through creator pattern - https://github.com/hyperledger/aries-framework-go/blob/master/pkg/framework/aries/default.go#L56. just a FYI.

Reply:
The problem is that an outbound dispatcher is needed for the messenger. But we create outbound dispatcher at the end.

Reply:
Yes, we have similar cases in default package - https://github.com/hyperledger/aries-framework-go/blob/master/pkg/framework/aries/default.go.